controllers/endpoint/endpoints_controller_test.go

```diff
@@ -150,9 +150,10 @@ func testWeederDedicatedEnvTest(t *testing.T) {
 	for _, test := range tests {
 		ctx, cancelFn := context.WithCancel(context.Background())
 		testEnv, reconciler := setupWeederEnv(ctx, t, test.apiServerFlags)
-		ns := testutil.CreateTestNamespace(ctx, g, reconciler.Client, strings.ToLower(test.name))
+		testNs := testutil.GenerateRandomAlphanumericString(g, 4)
```

Comment:
You do not need to use this utility function any longer. We should perhaps remove this now.
Use https://pkg.go.dev/k8s.io/apimachinery/pkg/util/rand#String instead.
Change this at all places where this utility function is used.